pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -147,6 +165,42 @@ func (p *adProvider) RefetchGroupPrincipals(principalID string, secret string) (
 	return groupPrincipals, err
 }
 
+func getUsernameFromObjectGUID(lConn *ldapv3.Conn, config *v32.ActiveDirectoryConfig, username string) (string, error) {
+	uuid := strings.TrimPrefix(username, "objectGUID=")
+
+	parsedGUID, err := guid.Parse(uuid)
+	if err != nil {
+		return "", fmt.Errorf("cannot parsing UUID: %w", err)
+	}
+
+	// search the user by its objectGUID
+	query := fmt.Sprintf(
+		"(&(%v=%v)(%s=%s))",
+		ObjectClass, config.UserObjectClass,
+		ObjectGUIDAttribute, guid.Escape(parsedGUID),
+	)
+	logrus.Debugf("LDAP Search query with objectGUID from UUID (%s): {%s}", parsedGUID.UUID(), query)
+
+	search := ldap.NewWholeSubtreeSearchRequest(
+		config.UserSearchBase,
+		query,
+		config.GetUserSearchAttributes(defaultUserAttributes...),
+	)
+
+	result, err := lConn.Search(search)
+	if err != nil {
+		return "", fmt.Errorf("LDAP search of user by objectGUID failed: %w", err)
+	}
+
+	if len(result.Entries) == 0 {
+		return "", errors.New("LDAP search of user by objectGUID returned no results")
+	}
+
+	// if found we can continue the login flow with the login attribute of the user
+	username = result.Entries[0].GetAttributeValue(config.UserLoginAttribute)
```

Comment:
You could just return this here rather than assigning to a temporary variable?

Reply:
The `username` variable already existed and it was overwritten, but IIRC it's a leftover from a previous implementation. Then I copy-pasted everything into this dedicated func, so that's probably the reason of it.

Changing it!